### PR TITLE
nancy.demo.async to project json

### PR DESCRIPTION
**Pull request by jchannon**
Building on top of #2155 

(Accidentally pushed to nancy repo and not my fork)

### CODE REVIEW

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+  </PropertyGroup>
+
+  <Import Project="$(VSToolsPath)\DNX\Microsoft.DNX.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>725549a7-98db-468d-870c-f1712bd0c0b7</ProjectGuid>
+    <RootNamespace>Nancy.Demo.Async</RootNamespace>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">..\..\..\artifacts\obj\$(MSBuildProjectName)</BaseIntermediateOutputPath>
+    <OutputPath Condition="'$(OutputPath)'=='' ">..\..\..\artifacts\bin\$(MSBuildProjectName)\</OutputPath>
```
Comment (khellang) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: I think this path points to a level above the solution?

File: samples/Nancy.Demo.Async/project.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿{
+  "version": "2.0.0-*",
+
+  "description": "Nancy is a lightweight web framework for the .Net platform, inspired by Sinatra. Nancy aim at delivering a low ceremony approach to building light, fast web applications.",
```
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: I don't think we need `description`, `authors`, `tags`, `projectUrl`, `licenseUrl`, and `iconUrl` for the samples?

File: samples/Nancy.Demo.Async/project.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "tags": [ "Nancy" ],
+  "projectUrl": "http://nancyfx.org",
+  "licenseUrl": "https://github.com/NancyFx/Nancy/blob/master/license.txt",
+  "iconUrl": "http://nancyfx.org/nancy-nuget.png",
+  "compilationOptions": {
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "Nancy": { "target": "project" },
+    "System.Net.Http": "4.0.1-beta-23516"
```
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: This is a framework assembly in .NET 4.5, right? Should probably be moved to the `frameworks` section.

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: Acually, I'm not sure we care about `version` either. These will never be packaged up and published, right?

Comment (jchannon) [inline samples/Nancy.Demo.Async/project.json]: version probably doesn't matter. Just for OCD might be best to keep authors, description etc

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+  </PropertyGroup>
+
+  <Import Project="$(VSToolsPath)\DNX\Microsoft.DNX.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>725549a7-98db-468d-870c-f1712bd0c0b7</ProjectGuid>
+    <RootNamespace>Nancy.Demo.Async</RootNamespace>
```
Comment (khellang) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: I think this can :fire:

File: samples/Nancy.Demo.Async/project.json (hunk 3 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/project.json]: Will try it, just using what VS recommended by pressing `Ctrl+.`

Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: Yeah, I wouldn't trust the tooling for this yet... :stuck_out_tongue_closed_eyes:

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj (hunk 4 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: what can?

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: They provide no value outside NuGet packages AFAIK. Anyway, if we're keeping them, they need sensible values.

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj (hunk 4 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: The `<RootNamespace>` :smile:

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/project.json]: Well I'm using your the values you provided :stuck_out_tongue:

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj (hunk 4 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: Again, this comes from File - New Project, what diff does it make as that is the namespace for the files currently present in the project

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: Yes, but those were carefully considered for each repo. These are just blatantly copy-pasted from the Nancy project.json :stuck_out_tongue:

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj (hunk 4 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: Sure, but do you always use what you get from "File - New Project" out of the box? I sure don't - which is why I didn't use it for any of the projects I ported.

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/project.json]: I think we can use what is there with the version removed. no specifics needed for the demos as the name should highlight what it does?

File: samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj (hunk 4 above)
Comment (jchannon) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: Yeah fair point but what's the issue with that namespace?

Comment (jchannon) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: Although removing it means I could copy paste the xproj for new dnx projects we want to get in

Comment (khellang) [inline samples/Nancy.Demo.Async/Nancy.Demo.Async.xproj]: That's exactly what I did. I just copy pasted a placeholder project.json and xproj, then used https://guidgenerator.com/ to create an individual GUID for each project and filled in the details for the project.json

File: samples/Nancy.Demo.Async/project.json (hunk 2 above)
Comment (khellang) [inline samples/Nancy.Demo.Async/project.json]: I'm submitting the rest without them. I'm not bothering coming up with sensible descriptions and tags for all the demo projects.

**Pull request closed without merging**